tests/test_analysis.py

```diff
@@ -1359,5 +1361,136 @@ def helper_test_update_pcb_modeling_props(analysis):
+        assert str(e) == "Update part modeling props error: CCA name is invalid."
+
+    try:
+        analysis.update_part_modeling_props("Test", "INVALID")
+        pytest.fail("No exception thrown when part modeling props is incorrect type.")
+    except SherlockUpdatePartModelingPropsError as e:
+        assert (
+            str(e) == "Update part modeling props error: "
+            "Part modeling props argument is invalid."
+        )
+
+    try:
+        analysis.update_part_modeling_props(
+            "Test",
+            {
+                "part_enabled": True,
+                "part_min_size": 1,
+                "part_min_size_units": "in",
+                "part_elem_order": "First Order (Linear)",
+                "part_max_edge_length": 1,
+                "part_max_edge_length_units": "in",
+                "part_max_vertical": 1,
+                "part_max_vertical_units": "in",
+                "part_results_filtered": True,
+            },
+        )
+        pytest.fail("No exception thrown when CCA name is missing.")
+    except SherlockUpdatePartModelingPropsError as e:
+        assert str(e) == "Update part modeling props error: CCA name is missing."
+
+    try:
+        analysis.update_part_modeling_props(
+            "Test",
+            {
+                "cca_name": "Card",
+                "part_min_size": 1,
+                "part_min_size_units": "in",
+                "part_elem_order": "First Order (Linear)",
+                "part_max_edge_length": 1,
+                "part_max_edge_length_units": "in",
+                "part_max_vertical": 1,
+                "part_max_vertical_units": "in",
+                "part_results_filtered": True,
+            },
+        )
+        pytest.fail("No exception thrown when part enabled is missing.")
+    except SherlockUpdatePartModelingPropsError as e:
+        assert str(e) == "Update part modeling props error: Part enabled is missing."
+
+    if not analysis._is_connection_up():
+        return
+
+    try:
+        analysis.update_part_modeling_props(
+            "Tutorial Project",
+            {
+                "cca_name": "Main Board",
+                "part_enabled": True,
+                "part_min_size": 1,
+                "part_min_size_units": "in",
+                "part_elem_order": "INVALID",
+                "part_max_edge_length": 1,
+                "part_max_edge_length_units": "in",
+                "part_max_vertical": 1,
+                "part_max_vertical_units": "in",
+                "part_results_filtered": True,
+            },
+        )
+        pytest.fail("No exception raised when using an invalid parameter.")
+    except Exception as e:
+        assert type(e) == SherlockUpdatePartModelingPropsError
+
+    try:
+        result = analysis.update_part_modeling_props(
+            "Tutorial Project",
+            {
+                "cca_name": "Main Board",
+                "part_enabled": False,
```

Comment:
Because the other fields are not required when setting part_enabled=False, it would be better not to include them. This will verify that the other fields are not required in this case.
Also, because the required fields are validated in previous tests, no need for another test like this that sets part_enabled=True.